Detections/OracleDatabaseAudit/OracleDBAuditConnectFromExternalIp.yaml

```diff
@@ -0,0 +1,34 @@
+id: 54aa2c17-acfd-4e3a-a1c4-99c88cf34ebe
+name: OracleDBAudit - Connection to database from external IP
+description: |
+  'Detects when connection to database is from external IP source.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: OracleDatabaseAudit
+    dataTypes:
+      - Syslog
+queryFrequency: 30m
+queryPeriod: 30m
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+  - Collection
+  - Impact
+query: |
+  OracleDatabaseAuditEvent
+  | where isnotempty(SrcIpAddr)
+  | where isnotempty(Action)
+  | where tolower(DbAction) == 'connect'

```

Comment:
Instead of using tolower() here, use `=~` which matches without case sensitivity.  This is better for performance of the query.

Reply:
updated